doc/mgmt/SONiC_Management_Framework_platform.md

```diff
@@ -0,0 +1,439 @@
+        |  |  +--rw oc-platform-psu:enabled?   boolean
+        |  +--ro state
+        |     +--ro oc-platform-psu:enabled?          boolean
+        |     +--ro oc-platform-psu:capacity?         oc-types:ieeefloat32
+        |     +--ro oc-platform-psu:input-current?    oc-types:ieeefloat32
+        |     +--ro oc-platform-psu:input-voltage?    oc-types:ieeefloat32
+        |     +--ro oc-platform-psu:output-current?   oc-types:ieeefloat32
+        |     +--ro oc-platform-psu:output-voltage?   oc-types:ieeefloat32
+        |     +--ro oc-platform-psu:output-power?     oc-types:ieeefloat32
+        +--rw fan
+        |  +--rw config
+        |  +--ro state
+        |     +--ro oc-pf-ext:direction?         string
+        |     +--ro oc-pf-ext:speed-tolerance?   uint32
+        |     +--ro oc-pf-ext:target-speed?      uint32
+        |     +--ro oc-fan:speed?                uint32
+```
+
+### 3.6.2 CLI
+There are only show commands for platform information.
+#### 3.6.2.1 Configuration Commands
+#### 3.6.2.2 Show Commands
+```
+sonic# show platform
+  environment  Show platform Environment
+  fanstatus    Show platform fan status
+  psustatus    Show platform PSU status
+  psusummary   Show platform PSU summary
+  syseeprom    Show platform EEPROM information
+  temperature  Show platform temperature sensors
+  |            Pipe through a command
+  <cr>
+```
+##### Show EERPOM
+```
+sonic# show platform syseeprom
+-----------------------------------------------------------
+Attribute            Value/State
+-----------------------------------------------------------
+description         :x86_64-dell_z9100_c2538-r0
+empty               :False
+hardware-version    :A02
+id                  :Z9100-ON
+location            :Slot 1
+mfg-date            :2017-08-07
+mfg-name            :CES00
+name                :System Eeprom
+base-mac-address    :34:17:EB:2C:D7:00
+diag-version        :3.23.4.1
+mac-addresses       :384
+manufacture-country :CN
+onie-version        :3.23.1.3
+service-tag         :847RG02
+vendor-name         :DELL
+oper-status         :ACTIVE
+part-no             :0KY5C4
+removable           :False
+serial-no           :CN0KY5C4CES007770009
+software-version    :'dell_sonic_3.x_share.129-6a2cad56d'
+
+```
+
+##### Show  PSU status
+```
+sonic# show platform psustatus
+-----------------------------------------------------------
+PSU                  Status
+-----------------------------------------------------------
+PSU 1                ACTIVE
+PSU 2                INACTIVE
+
+```
+##### Show PSU summary
+```
+sonic# show platform psusummary
+PSU 1:
+    description          02RPHX
+    mfg-name             DELL
+    fans                 1
+    status-led           green
+    oper-status          ACTIVE
+    serial-no            CN-02RPHX-17972-74S-018N-A00
+    output-current (W)   1.27
+    output-power (W)     127.50
+    output-voltage (V)   12.19
+    fan-speed (RPM)      5808
+    fan-direction        fan_direction_exhaust
```

Comment:
Hi Garrick,

The DB stores the airflow directions as "Intake" or "exhaust". As mentioned in the base class,
sonic-platform-common/sonic_platform_base/fan_base.py

# Possible fan directions (relative to port-side of device)
  FAN_DIRECTION_INTAKE = "intake"
  FAN_DIRECTION_EXHAUST = "exhaust" 

When I am printing the values in CLICK CLIs, I am printing the string as 'Intake' or 'Exhaust'. (using the capitalize() built-in method for strings). 

Can you use "Intake" or "Exhaust" here too instead of 'fan_direction_exhaust' or 'fan_direction_intake'?

Reply:
Hi @fk410167 
The DB actually stores airflow direction the way it's printed in the HLD:

```
admin@sonic:~$ redis-cli -n 6 HGETALL "FAN_INFO|FAN 7"
 1) "presence"
 2) "true"
 3) "status"
 4) "true"
 5) "speed"
 6) "6231"
 7) "target_speed"
 8) "0"
 9) "speed_tolerance"
10) "20"
11) "status_led"
12) "green"
13) "name"
14) "FanTray4-Fan1"
15) "direction"
16) "fan_direction_exhaust"         <=======
17) "serial"
18) "CN-03CH15-CES00-773-0016-A00"
19) "model"
20) "CN-03CH15-CES00-773-0016-A00"
admin@sonic:~$
```

Do you still want to change the output?

Reply:
I saw Dell platform's APIs and they return "fan_direction_exhaust" or "fan_direction_intake" which is written in the DB. This needs to be changed from Dell platform API side. 

Regardless, if you print the airflow string obtained from DB using str.capitalize(), then output would be uniform in CLICK and KLISH.

Reply:
Currently consulting the platform team on this. We usually represent our fan directions as "forward" and "reverse", or "front-to-back" and v/v. I brought this up in a couple of meetings. 

Anyway, it looks like this just prints whatever the platform API returns, so it could be an issue with what we're returning from the platform side and the dev team will follow up if there's an issue.